packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -221,7 +221,7 @@ export default class OptionManager {
 
     // resolve presets
     if (opts.presets) {
-      this.mergePresets(opts.presets, dirname);
+      this.mergePresets(opts.presets, dirname, opts.filename);
```

Comment:
Looking at this, couldn't this be solved more simply by passing the `package.json`'s dirname on https://github.com/babel/babel/blob/master/packages/babel-core/src/transformation/file/options/option-manager.js#L160 ?